interfaces/Portalicious/src/app/pages/users/add-user-form/add-user-form.component.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+
+import {
+  injectMutation,
+  injectQuery,
+} from '@tanstack/angular-query-experimental';
+import { ButtonModule } from 'primeng/button';
+import { DropdownModule } from 'primeng/dropdown';
+import { InputTextModule } from 'primeng/inputtext';
+import { MultiSelectModule } from 'primeng/multiselect';
+
+import { FormSidebarComponent } from '~/components/form/form-sidebar.component';
+import { FormFieldWrapperComponent } from '~/components/form-field-wrapper/form-field-wrapper.component';
+import { RoleApiService } from '~/domains/role/role.api.service';
+import { UserApiService } from '~/domains/user/user.api.service';
+import { ToastService } from '~/services/toast.service';
+import {
+  generateFieldErrors,
+  genericFieldIsRequiredValidationMessage,
+} from '~/utils/form-validation';
+
+type AddUserToTeamFormGroup =
+  (typeof AddUserFormComponent)['prototype']['formGroup'];
+
+@Component({
+  selector: 'app-add-user-form',
+  styles: ``,
+  standalone: true,
+  templateUrl: './add-user-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [
+    ButtonModule,
+    FormSidebarComponent,
+    FormFieldWrapperComponent,
+    DropdownModule,
+    MultiSelectModule,
+    InputTextModule,
+    ReactiveFormsModule,
+  ],
+})
+export class AddUserFormComponent {
+  // userToEdit = input<ProjectUserWithRolesLabel | undefined>();
+  formVisible = model.required<boolean>();
+
+  // private projectApiService = inject(ProjectApiService);
+  private userApiService = inject(UserApiService);
+  private roleApiService = inject(RoleApiService);
+  private toastService = inject(ToastService);
+
+  // constructor() {
+  //   effect(
+  //     () => {
+  //       const user = this.userToEdit();
+  //       if (user) {
+  //         this.formGroup.patchValue({
+  //           userValue: user.id,
+  //           rolesValue: user.roles.map(({ role }) => role),
+  //           scopeValue: user.scope,
+  //         });
+  //         this.formGroup.controls.userValue.disable();
+  //       } else {
+  //         this.formGroup.patchValue({
+  //           userValue: -1,
+  //         });
+  //         this.formGroup.controls.userValue.enable();
+  //       }
+  //     },
+  //     {
+  //       allowSignalWrites: true,
+  //     },
+  //   );
+  // }
+
+  // isEditing = computed(() => !!this.userToEdit());
+
+  roles = injectQuery(this.roleApiService.getRoles());
+  allUsers = injectQuery(this.userApiService.getAllUsers());
+
+  formGroup = new FormGroup({
+    usernameValue: new FormControl<string>('', {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required],
+      nonNullable: true,
+    }),
+    emailValue: new FormControl<string>('', {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      validators: [Validators.required, Validators.email],
```

Comment:
@arsforza RE: your comment:

> the email validator here on line 98 is never triggered because of the required validator
> should we remove the latter?
> I think the form would be invalid with an empty email field anyway

I don't see why it should not triggered. On the login page we have the same two validators, and they work correctly:

![image](https://github.com/user-attachments/assets/d5ef4675-0d1a-4175-b6dd-61d815398ef2)

Reply:
it's because it's a custom validation message on control.invalid for email
I added the same